ui/helpers/utils/export-utils.js

```diff
@@ -1,11 +1,93 @@
-import { getRandomFileName } from './util';
+/**
+ * @enum { string }
+ */
+export const ExportableContentType = {
+  JSON: 'application/json',
+  TXT: 'text/plain',
+};
 
-export function exportAsFile(filename, data, type = 'text/csv') {
+/**
+ * @enum { string }
+ */
+const ExtensionForContentType = {
+  [ExportableContentType.JSON]: '.json',
+  [ExportableContentType.TXT]: '.txt',
+};
+
+/**
+ * Export data as a file.
+ *
+ * @param {string} filename - The name of the file to export.
+ * @param {string} data - The data to export.
+ * @param {ExportableContentType} contentType - The content type of the file to export.
+ */
+export async function exportAsFile(filename, data, contentType) {
+  if (!ExtensionForContentType[contentType]) {
+    throw new Error(`Unsupported file type: ${contentType}`);
+  }
+
+  if (supportsShowSaveFilePicker()) {
+    // Preferred method for downloads
+    await saveFileUsingFilePicker(filename, data, contentType);
+  } else {
+    saveFileUsingDataUri(filename, data, contentType);
+  }
+}
+/**
+ * Notes if the browser supports the File System Access API.
+ *
+ * @returns {boolean}
+ */
+function supportsShowSaveFilePicker() {
+  return (
+    typeof window !== 'undefined' &&
+    typeof window.showSaveFilePicker !== 'undefined' &&
+    typeof window.Blob !== 'undefined'
+  );
+}
+
+/**
+ * Saves a file using the File System Access API.
+ *
+ * @param {string} filename - The name of the file to export.
+ * @param {string} data - The data to export.
+ * @param {ExportableContentType} contentType - The content type of the file to export.
+ * @returns {Promise<void>}
+ */
+async function saveFileUsingFilePicker(filename, data, contentType) {
+  const blob = new window.Blob([data], { contentType });
```

Comment:
Nit: Perhaps we should use `globalThis` instead? That should work in either the browser or Node.js. That might remove the necessity to mock this in tests, since `globalThis` exists on Node.js as well. Though that might require an ESLint config update as well.